src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -70,10 +71,10 @@ internal partial class AnalysisState
         private static readonly ObjectPool<Dictionary<int, DeclarationAnalyzerStateData>> s_currentlyAnalyzingDeclarationsMapPool = new ObjectPool<Dictionary<int, DeclarationAnalyzerStateData>>(() => new Dictionary<int, DeclarationAnalyzerStateData>(), PoolSize);
         private static readonly ObjectPool<PerAnalyzerState> s_perAnalyzerStatePool = new ObjectPool<PerAnalyzerState>(() => new PerAnalyzerState(s_analyzerStateDataPool, s_declarationAnalyzerStateDataPool, s_currentlyAnalyzingDeclarationsMapPool), PoolSize);
 
-        public AnalysisState(ImmutableArray<DiagnosticAnalyzer> analyzers, CompilationData compilationData)
+        public AnalysisState(ImmutableArray<DiagnosticAnalyzer> analyzers, CompilationData compilationData, bool trackPerAnalyzerState = true)
```

Comment:
Yes, that is correct. For GetAllDiagnosticsAsync API, where we need to return compiler + analyzer diagnostics, so the implementation is exactly matching the batch compiler. For GetAnalyzerDiagnosticsAsync and GetAnalyzerResultAsync we avoid invoking compilation.GetDiagnostics but instead use simulated event queue (we had seen VM issues with this approach in Update1) and then run analyzer driver without any state tracking.